pysilcam/silcamgui/interactive_summary.py

```diff
@@ -461,76 +498,87 @@ def toggle_plot(self):
 
     def update_plot(self, save=False):
         '''update the plots and save to excel is save=True'''
+
+        vd_total = xr.DataArray(data=self.vd_total, dims=['time', 'particle_size'], coords=[self.u, self.dias])
+        vd_gas = xr.DataArray(data=self.vd_gas, dims=['time', 'particle_size'], coords=[self.u, self.dias])
+        vd_oil = xr.DataArray(data=self.vd_oil, dims=['time', 'particle_size'], coords=[self.u, self.dias])
+
+        ds = xr.Dataset()
+        ds['vd_total'] = vd_total
+        ds['vd_gas'] = vd_gas
+        ds['vd_oil'] = vd_oil
+
         start_time = self.mid_time - self.av_window / 2
         end_time = self.mid_time + self.av_window / 2
-        u = pd.to_datetime(self.u)
-        timeind = np.argwhere((u >= start_time) & (u < end_time))
-
-        psd_nims = len(timeind)
+        
+        ds_slice = ds.sel(time=slice(start_time, end_time))
+        psd_nims = ds_slice.time.size
         if psd_nims < 1:
-            plt.sca(self.axispsd)
-            plt.cla()
-
-            plt.sca(self.axistext)
+            psd = ds_slice.mean(dim='time')
```

Comment:
F841: local variable 'psd' is assigned to but never used  <!-- Automatically posted by Lintly -->